src/model/class.ts

```diff
@@ -0,0 +1,246 @@
+    this.jsdoc = scannedClass.jsdoc;
+    this.description = scannedClass.description;
+    this.summary = scannedClass.summary;
+    this.abstract = scannedClass.abstract;
+    this.privacy = scannedClass.privacy;
+
+    if (scannedClass.superClass) {
+      this.superClass = scannedClass.superClass.resolve(document);
+    }
+    this.methods.push(...scannedClass.methods);
+    this.properties.push(...scannedClass.properties);
+
+    this._applySuperClass(scannedClass, document);
+    this._applyMixins(scannedClass, document);
+
+    for (const method of this.methods) {
+      // methods are only public by default if they're documented.
+      method.privacy = getOrInferPrivacy(method.name, method.jsdoc, true);
+    }
+  }
+
+  protected _inheritFrom(superClass: Class|ElementMixin) {
+    const existingProperties = new Set(this.properties.map((p) => p.name));
+    for (const superProperty of superClass.properties) {
+      if (existingProperties.has(superProperty.name)) {
+        continue;
+      }
+      const newProperty = Object.assign({}, superProperty);
+      newProperty.inheritedFrom = newProperty.inheritedFrom || superClass.name;
+      this.properties.push(newProperty);
+    }
+
+    const existingMethods = new Set(this.methods.map((m) => m.name));
+    for (const superMethod of superClass.methods) {
+      if (existingMethods.has(superMethod.name)) {
+        continue;
+      }
+      const newMethod = Object.assign({}, superMethod);
+      newMethod.inheritedFrom = newMethod.inheritedFrom || superClass.name;
+      this.methods.push(newMethod);
+    }
+  }
+
+  protected _applySuperClass(scannedElement: ScannedClass, document: Document) {
+    if (scannedElement.superClass &&
+        scannedElement.superClass.identifier !== 'HTMLElement') {
+      const superElements = document.getFeatures({
+        kind: 'class',
+        id: scannedElement.superClass.identifier,
+        externalPackages: true,
+        imported: true,
+      });
+      if (superElements.size === 1) {
+        const superClass = superElements.values().next().value;
+        this._inheritFrom(superClass);
+        return superClass;
+      } else {
+        if (superElements.size === 0) {
+          this.warnings.push({
+            message: `Unable to resolve superclass ` +
+                `${scannedElement.superClass.identifier}`,
+            severity: Severity.ERROR,
+            code: 'unknown-superclass',
+            sourceRange: scannedElement.superClass.sourceRange!,
+          });
+        } else {
+          this.warnings.push({
+            message: `Multiple superclasses found for ` +
+                `${scannedElement.superClass.identifier}`,
+            severity: Severity.ERROR,
+            code: 'unknown-superclass',
+            sourceRange: scannedElement.superClass.sourceRange!,
+          });
+        }
+      }
+    }
+    return undefined;
+  }
+
+  private _applyMixins(scannedClass: ScannedClass, document: Document) {
```

Comment:
I neglected to do this with `jsdoc.getMixins`, and `PolymerElement.inheritFrom`, but I think we need to reinforce the order that mixins apply in every function that deals with lists of mixins, because even I forget this. We also need to make sure this gets into docs.

The way we apply mixins @ master assumes we go from innermost (higher on the prototype chain, closer to Object) to outermost (lower, closer to concrete class). However I noticed in Polymer core that in the cases where there's more than one mixin they list from outermost to innermost.

I think the order that master assumes is the right one, because you can apply mixins as you read down the file - the mnemonic is "apply this mixin, then this mixin"...